Release resources ones client disconnects.

msisov opened the pull request:
This commit fixes a problem, when we have several ws::Display's and
client no longer needs to use it. Once it disconnected, we invalidate
frame sink id of the ws::Display's root and then ask display manager
to destroy displays (this ensures gpu related resources are down and
data is no longer flushed using not existing accelerated widgets).

Once we have just one ws::DIsplay, normal path is taken (Service destroys window_server, which destroys the last ws::Display).

tonikitoo reviewed (approved):
I think we should make it a fixup, if it is completes the logic in "Implement close request in external window mode."

we can ensure the proper authorship when we squash/rebase.

msisov committed:
fixup! Implement close request in external window mode.

This commit fixes a problem, when we have several ws::Display's and
client no longer needs to use it. Once it is disconnected, we invalidate
frame sink id of the ws::Display's root and then ask display manager
to destroy displays (this ensures gpu related resources are down and
data is no longer flushed using not existing accelerated widgets).

Once we have just one ws::Display, normal path is taken (Service
destroys window_server, which destroys the last ws::Display).
